router/delivery/http/router_handler.go

```diff
@@ -150,39 +148,66 @@ func (a *RouterHandler) GetOptimalQuote(c echo.Context) (err error) {
 
 // @Summary Compute the quote for the given poolID
 // @Description Call does not search for the route rather directly computes the quote for the given poolID.
+// @Description
+// @Description For exact amount in swap method, the `tokenIn` and `tokenOutDenom` are required.
+// @Description For exact amount out swap method, the `tokenOut` and `tokenInDenom` are required.
+// @Description Mixing swap method parameters in other way than specified will result in an error.
+// @Description
 // @ID get-direct-quote
 // @Produce  json
-// @Param  tokenIn         query  string  true  "String representation of the sdk.Coin for the token in."                   example(5OSMO)
-// @Param  tokenOutDenom   query  string  true  "String representing the list of the token denom out separated by comma."   example(ATOM,USDC)
-// @Param  poolID          query  string  true  "String representing list of the pool ID."                                  example(1,2,3)
+// @Param  tokenIn         query  string  false  "String representation of the sdk.Coin denoting the input token for the exact amount in swap method."                       example(1000000uosmo)
+// @Param  tokenOutDenom   query  string  false  "String representing the list of the output token denominations separated by comma for the exact amount in swap method."    example(uion)
+// @Param  tokenOut        query  string  false  "String representation of the sdk.Coin denoting the output token for the exact amount out swap method."                     example(2353uion)
+// @Param  tokenInDenom    query  string  false  "String representing the list of the input token denominations separated by comma for the exact amount out swap method."    example(uosmo)
```

Comment:
Would you mind adding `humanDenoms` functionality to this endpoint please? I think it would be a great dev UX improvement

Reply:
I have added `humanDenoms` including relevant tests covering this functionality. :+1: